### ISSUE TITLE
Source Address is not used in socket connection

### ISSUE THREAD

**Issue by rafaelrds** [cole/aiosmtplib#165]
At [`connection.py`](https://github.com/cole/aiosmtplib/blob/main/aiosmtplib/connection.py#L356), I believe the `source_address` should be used as `local_addr` parameter, otherwise socket `laddr` will always set to local IP address.
```python
...
        else:
            connect_coro = create_connection(
                self.loop,
                lambda: protocol,
                host=self.hostname,
                port=self.port,
                ssl=tls_context,
                ssl_handshake_timeout=ssl_handshake_timeout,
                local_addr=(self.source_address, 0),
            )
```

**Comment by cole** [comment]
Source address is a string sent to the SMTP server when connecting, identifying the client by IP or hostname (or any string, really). That's not very clear anywhere in the docs, but I don't think it's appropriate to use as `local_addr`.

**Comment by rafaelrds** [comment]
Yes @cole, it is for identifying the hostname and in our case, the sending machine can send e-mails for more than one domain, and has a few possible IPs when binding a socket. Let's say we are hosting from `1.2.3.4` but want to send from `1.2.3.5` if sending from domainA.com and `1.2.3.6` if it is from domainB.com (both are source addresses). I think adding a parameter to set `local_addr` might be the finest solution, so it can allow further customization. :1st_place_medal: 

Our current solution was to patch `_create_connection` method by creating a subclass of `SMTP`.
e.g.
```python
class MySMTP(SMTP):
    async def _create_connection(self) -> SMTPResponse:
        ...
                else:
                    connect_coro = create_connection(
                        self.loop,
                        lambda: protocol,
                        host=self.hostname,
                        port=self.port,
                        ssl=tls_context,
                        ssl_handshake_timeout=ssl_handshake_timeout,
                        local_addr=(self._source_address, 0),
            )
        ...
```

As for the use case, some Email Service Providers defers messages when the `client-ip` diverges from the IP presented at `EHLO`.

**Comment by rafaelrds** [comment]
Hey @cole. I've notice that the official CPython `smtplib` does use the source_address/hostname when creating a socket during `connect()` https://github.com/python/cpython/blob/master/Lib/smtplib.py#L339.

**Comment by cole** [comment]
@rafaelrds ah, that's a good point. A better approach may be to use `source_address` after all then.

**Comment by cole** [comment]
Closed by #167
